pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -111,10 +133,18 @@ func (cloud *CloudProvider) InsertDisk(ctx context.Context, zone string, diskToC
 	return cloud.service.Disks.Insert(cloud.project, zone, diskToCreate).Context(ctx).Do()
 }
 
+func (cloud *CloudProvider) CreateSnapshot(ctx context.Context, zone string, diskName string, snapshotToCreate *compute.Snapshot) (*compute.Operation, error) {
+	return cloud.service.Disks.CreateSnapshot(cloud.project, zone, diskName, snapshotToCreate).Context(ctx).Do()
```

Comment:
`waitForGlobalOp` should be done inside the cloud provider calls. I know currently the other calls are not like this, but I am currently changing it to be like that in order to be more consistent with Kubernetes code pattersn